core/src/main/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTask.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.dsp.workbench.welder
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect.{IO, Timer}
+import fs2.Stream
+import io.chrisdavenport.log4cats.Logger
+
+import scala.concurrent.duration.FiniteDuration
+
+object BackgroundTask {
+  def cleanUpLock(metadataCache: MetadataCache, interval: FiniteDuration)(
+      implicit logger: Logger[IO],
+      timer: Timer[IO]
+  ): Stream[IO, Unit] = {
+    val task = (for {
+      now <- timer.clock.monotonic(TimeUnit.MILLISECONDS)
+      updatedMap <- metadataCache.modify { mp =>
+        val newMap = mp.map { kv =>
+          val newLock = kv._2.lock.filter(l => l.lockExpiresAt.toEpochMilli < now)
+          (kv._1 -> kv._2.copy(lock = newLock))
+        }
+        (newMap, newMap)
+      }
+      _ <- logger.info(s"updated metadata cache ${updatedMap.take(100)}")
```

Comment:
Wondering if it's ok to log the cache... we don't want to inadvertently fill up the disk (that the user is paying for) with these log messages. Maybe we could log something like `"updated metadata cache with n entries"` at INFO level and log the actual cache at DEBUG level.

Reply:
Good idea..updated